File: tests/snapshots/asc_iif.rs
// Two classic VB builtins: Asc (the inverse of Chr) and IIf (immediate-if).
// Asc gives a character's code; IIf picks one of two values by a condition
// (lowered to a lazy Rust `if`/`else`, so — unlike VB — only one arm runs).

fn main() {
    println!("{}", "A".chars().next().map_or(0, |c| c as i64));
    // 65
    println!("{}", ((("A".chars().next().map_or(0, |c| c as i64) + 1) as u8) as char).to_string());
    // "B" — next letter
    let size: String = (if 10 > 3 { "big" } else { "small" }).to_string();
    println!("{}", size);
    let n: i64 = (if 4 % 2 == 0 { 100 } else { 200 }) as i64;
    println!("{}", n);
    // Mismatched arms (an owned String and a &str literal) still unify.
    let word: String = "hello".to_string();
    println!("{}", (if "z".chars().next().map_or(0, |c| c as i64) > 100 { word } else { "?".to_string() }));
}

File: tests/snapshots/asc_iif.py
# Two classic VB builtins: Asc (the inverse of Chr) and IIf (immediate-if).
# Asc gives a character's code; IIf picks one of two values by a condition
# (lowered to a lazy Rust `if`/`else`, so — unlike VB — only one arm runs).

def _vb(x):
    if isinstance(x, bool):
        return "true" if x else "false"
    if isinstance(x, float) and x.is_integer():
        return str(int(x))
    return str(x)

def main():
    print(_vb((ord(('A')[0]) if ('A') else 0)))
    # 65
    print(_vb(chr(int((ord(('A')[0]) if ('A') else 0) + 1) & 255)))
    # "B" — next letter
    size: str = ('big') if (10 > 3) else ('small')
    print(_vb(size))
    n: int = (100) if ((4 % 2) == 0) else (200)
    print(_vb(n))
    # Mismatched arms (an owned String and a &str literal) still unify.
    word: str = 'hello'
    print(_vb((word) if ((ord(('z')[0]) if ('z') else 0) > 100) else ('?')))


if __name__ == "__main__":
    main()
